### DISCUSSION TITLE
📊 Vote for New Database Integrations for Wren AI

### DISCUSSION THREAD

**Discussion by wwwy3y3** [Canner/WrenAI#327] [General]
Hi everyone,

We’re excited to gather your input on which database integrations you’d like to see next in WrenAI! Your feedback is crucial in helping us prioritize and focus on what’s most valuable for our community.

Please take a moment to review the list of potential database integrations below and vote for the ones you find most useful. If you have any additional suggestions, feel free to add them in the comments.

## How to Vote:
Add a thumbs-up (👍) reaction to the comment(s) corresponding to the database integration(s) you support.
If you have a new suggestion, please add a new comment and describe the integration.

**Comment by cyyeh** [comment]
(Update) 0.8.0 has supported Trino (https://github.com/Canner/WrenAI/releases/tag/0.8.0)

ref: https://github.com/Canner/WrenAI/issues/223

**Reply by wwwy3y3** [reply]
Trino has been scheduled to be supported in the following releases as the forth priority, coming after MySQL, MS SQL Server & Clickhouse.

Thanks for everyone's feedback.

**Reply by ng22t7a** [reply]
can u tell when will trino be supported

**Reply by cyyeh** [reply]
@ng22t7a Trino is supported now

**Reply by chilijung** [reply]
@ng22t7a check docs here https://docs.getwren.ai/oss/guide/connect/trino

**Reply by akash247777** [reply]
waiting for MongoDB support

**Comment by cyyeh** [comment]
**(Update) 0.6.0 has supported MS SQL Server (https://github.com/Canner/WrenAI/releases/tag/0.6.0)**

MS SQL Server

ref: https://github.com/Canner/WrenAI/issues/173

**Reply by wwwy3y3** [reply]
MS SQL Server has been scheduled to be supported in the following releases as the second priority, coming after MySQL.

Thanks for everyone's feedback.

**Reply by cyyeh** [reply]
Guys, we now support MS SQL Server in our latest version! https://github.com/Canner/WrenAI/releases/tag/0.6.0
Please have a try and we look forward to your feedback ❤️

**Comment by cyyeh** [comment]
OracleDB

ref: https://github.com/Canner/WrenAI/issues/236

**Reply by Redhadjeridi** [reply]
any updates about the oracle DB?

**Reply by numberonewastefellow** [reply]
Basically oracle PR merged, but it doesn't support fully. couple of places oracle implementations are missed.

**Reply by chilijung** [reply]
This is supported in 0.23.0 version!

**Comment by cyyeh** [comment]
**(Update) 0.5.0 has supported MySQL (https://github.com/Canner/WrenAI/releases/tag/0.5.0)**

MySQL

ref: https://github.com/Canner/WrenAI/issues/313

**Reply by wwwy3y3** [reply]
MySQL has been scheduled to be supported in the next release as the first priority.

Thanks for everyone's feedback.

**Reply by onlyjackfrost** [reply]
Guys, we now support MySQL in our latest version! https://github.com/Canner/WrenAI/releases/tag/0.5.0
Have a try and thanks for all your's feedback ❤️

**Reply by jagomezdiez** [reply]
NO, Invalid value 'mysql'

**Comment by cyyeh** [comment]
Databricks SQL Database/Lakehouse

ref: https://github.com/Canner/WrenAI/issues/322

**Reply by federel** [reply]
is databricks sql in your roadmap?

**Reply by emilianomm** [reply]
Are you planning to integrate with databricks using sql jdbc/odbc drivers or integrate nativey with Unity Catalog?

**Reply by Haybu** [reply]
looking forward for Databricks support

**Reply by chilijung** [reply]
Databricks is now supported: https://docs.getwren.ai/oss/guide/connect/databricks

**Comment by qdrddr** [comment]
# Neo4J Knowledge Graph DB
It can serve as Vector DB, or as SQL alternative for table relational structured data using Cypher query language.
Making it ideal and notable candidate for AI use-cases such as WrenAI.

Models such as OpenAI/gpt3.5-turbo and ollama with these models
https://ollama.com/library/sqlcoder
https://ollama.com/library/codeqwen
https://ollama.com/library/starcoder2
https://ollama.com/library/llama3:8b-instruct

All are quite good at generating Cypher query with prompt like this:
[cypher_generation_request4.txt](https://github.com/Canner/WrenAI/files/15473001/cypher_generation_request4.txt)

@cyyeh

**Reply by cyyeh** [reply]
@qdrddr interesting, I agree that somehow graph database is more suitable for modeling relational data than vector db. Also it's a big plus if graph database also supports vector semantic search. I am testing another graph database called [Nebula](https://www.nebula-graph.io/)(based on Apache 2) in my free time lol.

reference implementation: https://github.com/Canner/WrenAI/tree/poc/ai-service/graphdb-as-document-store

**Reply by qdrddr** [reply]
Did you know that Neo4j and Nebula both supports the Cypher query language?
https://docs.nebula-graph.io/2.0/3.ngql-guide/1.nGQL-overview/1.overview/

By the way there are initiatives to help with Text2Cypher creation (using LLMs)

https://bratanic-tomaz.medium.com/crowdsourcing-text2cypher-dataset-e65ba51916d4?source=post_page-----4cd4927b84f8--------------------------------

This one is notable framework that focuses on building automated knowledge graphs. Extending, and query KGs. That you might want to know of @cyyeh and maybe you'll be interested in implementing in WrenAI
https://github.com/whyhow-ai/whyhow

**Reply by cyyeh** [reply]
@qdrddr yeah, thanks for your resources. I've also seen these few days ago haha.

**Reply by qdrddr** [reply]
A few useful links about text2cypher
https://github.com/easwee/text2cypher
https://github.com/neo4j-labs/text2cypher
https://siwei.io/en/llm-text-to-nebulagraph-query/
https://ollama.com/ahuguenardlogility/text2cypher-q4
https://ollama.com/tomasonjo/llama3-text2cypher-demo
https://www.nebula-graph.io/posts/build-knowledgegraph-via-natural-language-demo
https://www.nebula-graph.io/posts/graph-rag-demo

https://github.com/neo4j-labs/text2cypher/tree/main/finetuning/unsloth-llama3#using-chat-prompt-template

HuggingFace model:https://huggingface.co/tomasonjo/text2cypher-demo-16bit

Ollama models: https://ollama.com/tomasonjo/llama3-text2cypher-demo

**Reply by cyyeh** [reply]
@qdrddr are u interested in integrating graph database in WrenAI? would like to give it a try haha? Currently we are busy implementing our first version of evaluation framework: https://github.com/Canner/WrenAI/issues/355

After it's done, I think it'll become a lot easier for us to refine ai pipelines along the way :)

**Reply by cyyeh** [reply]
If u are interested, maybe we can discuss first on what might be the next directions to go along?

**Reply by qdrddr** [reply]
Sure!

**Reply by qdrddr** [reply]
I think these are the directions:
- Visualization is different for Graph DB. Though you need to visualize it so users can enter metadata similarly as you are doing it for SQL to help LLM better understand your data.
- Graph DB can work with tabular data such as CSV but also you can extract Schema/Ontology from other types of data such as text from a book.
- Test that the models can produce working Cypher queries with a given prompt template and examples. Feel free to use mine [cypher_generation_request4.txt](https://github.com/Canner/WrenAI/files/15473001/cypher_generation_request4.txt)
- Another important direction unlike CSV / Relational DB is to produce Schema/Ontology for your Graph DB, is something users will be looking for.

**Reply by cyyeh** [reply]
Now my thoughts on using graphdb is to test whether it can replace qdrant. For visualization part, I agree it's important, but for WrenAI users, I think it might be implementation details, and they don't need to know. For WrenAI developers, there is corresponding UI that can help visualize the graph directly. Nebula also has one, and I've added it in the branch I mentioned earlier.

**Reply by qdrddr** [reply]
Yep, its possible to replace qdrant with Neo4j for Vectorized data.

**Reply by qdrddr** [reply]
> Now my thoughts on using graphdb is to test whether it can replace qdrant. For visualization part, I agree it's important, but for WrenAI users, I think it might be implementation details, and they don't need to know. For WrenAI developers, there is corresponding UI that can help visualize the graph directly. Nebula also has one, and I've added it in the branch I mentioned earlier.

As for the UI part, it depends. You might still want to add some metadata to the KG like you already doing it in WrenUI for the SQL Schema. So Maybe you can do it via Neo4j Browser UI if not then you'll need this UI in WrenAI.

**Reply by cyyeh** [reply]
My thought is users still edit metadata in UI of WrenAI which will make changes to MDL; then when MDL is deployed, wren-ai-service will take care of the rest, including adding metadata in MDL to KG

**Reply by qdrddr** [reply]
Sound like a plan!

**Reply by cyyeh** [reply]
@qdrddr falkordb is also an interesting target! It's based on redis

References
- https://www.falkordb.com/
- https://www.falkordb.com/blog/surevy-graphrag-and-knowledge-graphs-in-large-language-models/

**Reply by qdrddr** [reply]
LangChain has ability to generate Cypher queries now
https://github.com/tomasonjo/blogs/blob/master/llm/langchain_neo4j.ipynb

**Reply by qdrddr** [reply]
Also I came across the idea of building a semantic search layer between LLM and Graph DB. I guess the idea is very similar to what  WrenAI is already doing for SQL/Relational DB.

https://towardsdatascience.com/enhancing-interaction-between-language-models-and-graph-databases-via-a-semantic-layer-0a78ad3eba49

**Reply by qdrddr** [reply]
Its cool that FalkorDB also supports openCypher. It seams like Cypher become de-facto standard for Graph DB just like SQL.

**Reply by qdrddr** [reply]
there's another one, though it doesn't have Cypher, instead it supports its own "GraphQL-inspiered", which is a downside.
https://github.com/dgraph-io/dgraph

And also 
https://github.com/surrealdb/surrealdb
https://github.com/janusgraph/janusgraph

**Reply by qdrddr** [reply]
SurrealDB seams to be most popular among these Graph Databases.
Also there's a very popular Supabase Database, it based on Postgresql with GraphQL extension support, though not fully fledged Graph Database.

https://star-history.com/#neo4j/neo4j&supabase/supabase&surrealdb/surrealdb&dgraph-io/dgraph&vesoft-inc/nebula&falkordb/falkordb&janusgraph/janusgraph&Timeline

**Reply by qdrddr** [reply]
Here's an idea for you for KGDB: at the modeling stage, based on data and schema generate in advance all possible (at least primitive) queries that are checked and working (returning at least one row), and save into Vector DB, so you can retrieve examples dynamically based on the prompt. @cyyeh

**Reply by TheBestOrNothing** [reply]
janusGraph please

**Comment by thaddeuscleo** [comment]
Support for:
- Clickhouse

**Reply by wwwy3y3** [reply]
Hi @thaddeuscleo thanks for the suggestion. Since we already got `MySQL` as an option above, could we only keep `Clickhouse` in this comment ?

Thanks.

**Reply by thaddeuscleo** [reply]
Okay Sure, I'll update it @wwwy3y3

**Reply by qdrddr** [reply]
Actually, I do not see MySQL as an option anywhere else. @wwwy3y3

**Reply by wwwy3y3** [reply]
@qdrddr it's here -> https://github.com/Canner/WrenAI/discussions/327#discussioncomment-9578181

**Reply by wwwy3y3** [reply]
Clickhouse has been scheduled to be supported in the following releases as the third priority, coming after MySQL & MS SQL Server.

Thanks for everyone's feedback.

**Reply by maincrouser** [reply]
How can I use the ClickHouse connection with an SSL certificate? There is an option to enable SSL, but no possibility to choose a certificate.

**Comment by fabricebaranski** [comment]
Apache Druid

**Comment by MasaruKitajima** [comment]
MongoDB support will be appreciated.

**Comment by HelenGuohx** [comment]
Snowflake

**Reply by ongdisheng** [reply]
@HelenGuohx Snowflake is supported now, you may check the docs [here](https://docs.getwren.ai/oss/guide/connect/snowflake).

**Comment by qdrddr** [comment]
Supabase DB

**Comment by qdrddr** [comment]
FalkorDB Graph DB

**Comment by qdrddr** [comment]
SurrealDB Graph DB

**Comment by qdrddr** [comment]
Nebula Graph DB

**Comment by vietnguyengit** [comment]
ZARR store, https://zarr.readthedocs.io/en/stable/getting_started.html

**Reply by wwwy3y3** [reply]
@vietnguyengit thanks for the feedback! It's the first time I've heard of ZARR.

Could you let us know how you use ZARR store and why you think it's important to support? thanks!

**Comment by imtarundhiman** [comment]
MongoDB

**Comment by ragingbal** [comment]
Apache Doris

**Comment by prav3929** [comment]
Teradata

**Comment by tomthebuzz** [comment]
Doris &
DB2

**Comment by mayurp-yb** [comment]
Support for YugabyteDB

**Comment by SurenderSardana99** [comment]
Databrick Sql

**Comment by SurenderSardana99** [comment]
Any Data base without ER

**Comment by AnishNair94** [comment]
Apache Pinot.

Using pinot as olap store and for user facing reporting interface. Wren AI will be really awesome tool to integrate with User Reporting Interface @cyyeh

**Comment by samantagupta29** [comment]
AWS Athena

**Comment by avirup48** [comment]
Snowflake

**Reply by ongdisheng** [reply]
@avirup48 Snowflake is supported now, you may check the docs [here](https://docs.getwren.ai/oss/guide/connect/snowflake).

**Comment by johnfelipe** [comment]
SAP
SAP Ariba Procurement
SAP Ariba Source
SAP Business One
Business Objects BI
SAP ByDesign
SAP Concur
SAP Fieldglass
SAP Gateway
SAP HANA
SAP HANA XS Advanced
SAP Hybris C4C
SAP Netweaver Gateway
SAP SuccessFactors

**Comment by MGomaa435** [comment]
Add support to generic database driver, and also adding different data virtualization connectors like tibco data virtualization and, denodo.

**Comment by axoxymous-coder** [comment]
Amazon Redshift

**Reply by pellepelle3** [reply]
Can we use the postgres connector for this?

**Reply by pellepelle3** [reply]
I tried but the CREATE TEMPORARY VIEW doesn't exist it is instead just # in front of the view name

**Comment by federel** [comment]
Azure Data Explorer (ADX Kusto KQL)

**Comment by J9527H** [comment]
AWS Redshift

**Comment by mjmjmmmjjj** [comment]
ORACLE

**Comment by Coysh** [comment]
Dataverse / Microsoft Dynamics

**Comment by MyAIPlugins** [comment]
Astra DB

**Comment by funkyfreestyler** [comment]
PrestoDB.

**Comment by huangqz71** [comment]
StarRocks

**Reply by kwontaeheon** [reply]
Need it

**Comment by dduran28** [comment]
Amazon Redshift!

**Reply by pellepelle3** [reply]
> Amazon Redshift!

With some hacks to ibis I got postgres to work with redshift but would love to get that upsteam

**Reply by flapitgo** [reply]
@pellepelle3 I'm also interested in using Amazon Redshift. Can you please share what you did in ibis to make the postgres connector to work? Thanks

**Reply by alexlichtenstein** [reply]
@pellepelle3 @flapitgo can you share these hacks?

**Comment by gowridev** [comment]
@wwwy3y3 Could you please create integration plugin for TDengine(https://tdengine.com/). Its columnar time series database for lOT applications. its supports high scalability. 
https://github.com/taosdata/TDengine
I think, it will be 1st scalable Time series database integration with WRENAI

**Comment by philliprossi** [comment]
LookML

**Reply by dduran28** [reply]
this is interesting, are you saying an integration that will allow wren to use LookML semantic layer

**Comment by dgaruolis** [comment]
Apache Impala

**Comment by jonathanyb** [comment]
It is essential to inherit the stored procedure and query functionalities of the database, as certain complex data generation processes heavily rely on these features. Storing procedures and queries to generate new tables for the results of specialized queries is both frequent and crucial in real-world business operations. Without such capabilities, relying solely on simple cross-table queries would significantly limit operational efficiency and practicality. While the original statement may appear somewhat blunt, it accurately reflects the realities of corporate business scenarios.

**Comment by jiezi2026** [comment]
Apache Doris

**Comment by jiezi2026** [comment]
JDBC URL

**Reply by emilianomm** [reply]
yeah dont cut our wings

**Comment by patricklenert** [comment]
LibSQL (Turso)

**Comment by akashdeep01** [comment]
Apache Pinot

**Reply by abhishek-msh** [reply]
is there any integration?

**Comment by juneDink** [comment]
file system

**Comment by pierre-H** [comment]
Convex : https://convex.dev

**Comment by robertalv** [comment]
Convex

**Comment by javalxs** [comment]
damengdb

**Comment by ShubhamTyagiZepto** [comment]
StarRocks https://www.starrocks.io/

**Comment by saurabhmbai** [comment]
Medplum is an open-source FHIR server and healthcare data platform that enables secure, structured access to clinical data using the HL7 FHIR standard. Integrating Medplum with Wren AI allows GenAI agents to retrieve and reason over real-world medical records, enabling accurate, compliant, and context-aware healthcare insights.

https://www.medplum.com

**Comment by hahau6** [comment]
apache doris

**Reply by PainAshura** [reply]
+1

**Comment by PainAshura** [comment]
Apache Doris

**Comment by leap233** [comment]
Apache Doris 
please !!!

**Comment by tmotyl** [comment]
SQLite

**Comment by mesbahkhan** [comment]
InfluxDB

**Comment by ijtarano** [comment]
Dremio MCP https://github.com/dremio/dremio-mcp/tree/main

**Comment by nhtha** [comment]
Hello everyone,
Any plan for full-text-search solutions (eg. ELK) support?
Regards,
Ha Nguyen

**Comment by abdelhameedhamdy** [comment]
SQLite

**Reply by tmotyl** [reply]
aleady requested https://github.com/Canner/WrenAI/discussions/327#discussioncomment-13420858

**Comment by JinCheng666** [comment]
达梦数据库管理系统 DM8

**Comment by oankit** [comment]
Salesforce

**Comment by oankit** [comment]
Databricks

**Comment by brunabaldasso98** [comment]
Cognite

**Comment by shanebelgal1992** [comment]
Redshift

**Comment by deepaksaipendyala** [comment]
MongoDB

**Reply by vinay-berrydb** [reply]
Mongo support has already been requested, by doing a simple find in page(ctrl+f) before posting you would have known.

**Reply by deepaksaipendyala** [reply]
Thank you for doing ctrl+f, much appreciated.

**Comment by vinay-berrydb** [comment]
Couchbase

**Comment by k-alnuaimi** [comment]
Azure Log Analytics

**Comment by kalatalo** [comment]
Intersystems IRIS

**Comment by datalee** [comment]
excel

**Comment by bowenhiretual** [comment]
Apache Pinot / StarTree

**Comment by tradercjz** [comment]
DolphinDB

**Comment by andre9333** [comment]
4D 
https://developer.4d.com/
🙏🙏

**Comment by jxtanchris** [comment]
Filemaker

**Comment by diogolr** [comment]
SAP HANA DB

**Comment by DereC4** [comment]
DerexXD 👍

**Comment by pramodbhatofficial** [comment]
Databricks

**Reply by chilijung** [reply]
Databricks is now supported: https://docs.getwren.ai/oss/guide/connect/databricks

**Comment by xiangqian0011** [comment]
hive

**Comment by xiangqian0011** [comment]
hive

**Comment by Wkzlike** [comment]
Vertica Database

**Comment by MarioDeFelipe** [comment]
SAP HANA (I can help here)

**Comment by francescoscalzo** [comment]
elasticsearch

**Comment by selamar** [comment]
MonetDB

**Comment by uncixii** [comment]
Doris

**Comment by harrydevforlife** [comment]
starrocks.

I think it's suitable for analytic purposes.

**Comment by jucas65** [comment]
I would like it to be able to connect to Cube.js (https://cube.dev/). It is also open source and is a tool that allows you to create OLAP cubes on the fly from any data source, exposing them via API or SQL as if it were a PostgreSQL database. I tried connecting WrenAI, but Cube.js only supports SELECT queries and does not allow CREATE operations or similar.

**Comment by stevesutton-hds** [comment]
I was only tempted to install this, to "play" with it as I investigate approprate processing for my company, but all of our data is in AWS/dynamodb, and we'd need support for that, for me to even begin any testing.  Not a priority here, however, as I'm going to move on to other things.

**Comment by taoistwar** [comment]
hive

**Comment by zhyyyq** [comment]
ibm db2

**Comment by tusharvb19** [comment]
Time Series databases - influxDB, TimeScaleDB, TDengine

**Comment by minhocean** [comment]
Dremio

**Comment by jinghouyu** [comment]
HiveSQL

**Comment by wangyujia2023** [comment]
doris

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by medetshatayev** [comment]
Greenplum

**Comment by FanoShannon** [comment]
StarRocks please！
